### ISSUE TITLE
Underlying exception message is lost if SQL state code is unknown

### ISSUE THREAD

**Issue by sgrimm** [jOOQ/jOOQ#12708]
### Expected behavior

If a database operation fails, jOOQ should throw an exception with a meaningful error message, even if it can't categorize the failure.

### Actual behavior

For some database errors, the underlying error message from the database is discarded and you get an "Unspecified RuntimeException" message.

### Steps to reproduce the problem

I've run into this in a few contexts, most recently with PostGIS. For example, ask it to parse a malformed GeoJSON value:

```kotlin
    dslContext.query("SELECT ST_GeomFromGeoJson('{\"type\":\"Point\"}')").execute();
```

This fails with `org.jooq.exception.DataAccessException: SQL [SELECT ST_GeomFromGeoJson('{}')]; Unspecified RuntimeException`.

If I set a breakpoint in `JooqExceptionTranslator.exception()`, I see that the JDBC driver is throwing an exception with an informative error message: `org.postgresql.util.PSQLException: ERROR: Unable to find 'coordinates' in GeoJSON string`.

Tracing through the exception translation code (caveat: this is the first time I've read that code) it seems like the problem is that the exception has a `sqlState` value of `XX000` which isn't in the list of recognized states. So `SQLErrorCodeSQLExceptionTranslator.doTranslate()` is translating it to null.

### Versions

- jOOQ: 3.15.4
- Java: OpenJDK 15.0.1
- Database (include vendor): PostgreSQL 13.4 with PostGIS 3.1.3
- OS: MacOS 12.0.1
- JDBC Driver (include name if inofficial driver): 42.2.24

**Comment by lukaseder** [comment]
I'm assuming this is `org.springframework.boot.autoconfigure.jooq.JooqExceptionTranslator` and `org.springframework.jdbc.support.SQLErrorCodeSQLExceptionTranslator`? Can you report the bug to Spring, then?

With standalone jOOQ, I'm getting the expected stack trace:

```
08:46:39,946 DEBUG [LoggerListener                ] - Executing query          : SELECT ST_GeomFromGeoJson('{"type":"Point"}')
08:46:40,016 DEBUG [LoggerListener                ] - Exception                
org.jooq.exception.DataAccessException: SQL [SELECT ST_GeomFromGeoJson('{"type":"Point"}')]; ERROR: Unable to find 'coordinates' in GeoJSON string
	at org.jooq_3.16.0-SNAPSHOT.POSTGRES_14.debug(Unknown Source) ~[?:?]
	at org.jooq.impl.Tools.translate(Tools.java:3081) ~[classes/:?]
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:670) ~[classes/:?]
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:349) [classes/:?]
	at org.jooq.testscripts.JDBC.main(JDBC.java:40) [test-classes/:?]
Caused by: org.postgresql.util.PSQLException: ERROR: Unable to find 'coordinates' in GeoJSON string
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2657) ~[postgresql-42.3.0.jar:42.3.0]
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2347) ~[postgresql-42.3.0.jar:42.3.0]
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:337) ~[postgresql-42.3.0.jar:42.3.0]
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:484) ~[postgresql-42.3.0.jar:42.3.0]
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:404) ~[postgresql-42.3.0.jar:42.3.0]
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:162) ~[postgresql-42.3.0.jar:42.3.0]
	at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:151) ~[postgresql-42.3.0.jar:42.3.0]
	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219) ~[classes/:?]
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:414) ~[classes/:?]
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:335) ~[classes/:?]
	... 1 more
0
ERROR: Unable to find 'coordinates' in GeoJSON string
XX000
```

**Comment by sgrimm** [comment]
For anyone else who runs into this problem: this was indeed a Spring Boot bug, and it was fixed in Spring Boot 2.4.4.

**Comment by lukaseder** [comment]
Thanks a lot for documenting this, @sgrimm
